plugins/jvm/ts/discover.ts

```diff
@@ -4,28 +4,31 @@ namespace JVM {
 
   _module.controller("JVM.DiscoveryController", ["$scope", "localStorage", "jolokia", ($scope, localStorage, jolokia) => {
 
-    $scope.discovering = true;
-    $scope.agents = <any>undefined;
+    $scope.jvmDiscovery = {
+      discovering : true,
+      responseJson : undefined,
+      agents : <any>undefined,
+      selectedAgent: undefined,
+      filter: undefined
+    };
 
-    $scope.$watch('agents', (newValue, oldValue) => {
+    $scope.$watch('jvmDiscovery.agents', (newValue, oldValue) => {
       if (newValue !== oldValue) {
-        $scope.selectedAgent = _.find($scope.agents, a => a['selected']);
+        $scope.jvmDiscovery.selectedAgent = _.find($scope.jvmDiscovery.agents, a => a['selected']);
       }
     }, true);
 
     $scope.closePopover = ($event) => {
-      (<any>$)($event.currentTarget).parents('.popover').prev().popover('hide');
+      (<any>$)($event.currentTarget).parents('.popover').hide();
     };
 
     function getMoreJvmDetails(agents) {
       for (let key in agents) {
         const agent = agents[key];
         if (agent.url && !agent.secured) {
-          const dedicatedJolokia = createJolokia(Core.useProxyIfExternal(agent.url), agent.username, agent.password);
+          const dedicatedJolokia = createJolokia(agent.url, agent.username, agent.password);
```

Comment:
Why is usage of `Core.useProxyIfExternal(..)` removed?

Reply:
Good catch, that wasn't meant to be pushed! Will revert this.

Reply:
I reverted it ,  but then I noticed that the function useProxyIfExternal strips the protocol when building the proxy URL.

It yields a HTTP 406 with following URL: 
`http://127.0.0.1:8080/integration/proxy/localhost/8084/actuator/jolokia/read/java.lang%3Atype%3DRuntime/SystemProperties/sun.java.command?ignoreErrors=true&canonicalNaming=false`

Where we should expect the protocol *http* to be part of the path:
`http://127.0.0.1:8080/integration/proxy/http/localhost/8084/actuator/jolokia/read/java.lang:type=Runtime/SystemProperties/sun.java.command?ignoreErrors=true&canonicalNaming=false`

I don't want to toy around the Hawtio core functions, do you have any idea why the protool is being stripped ?


```js
            // hawtio-core.js line 4379
            // lets remove the http stuff
            var idx = connectUrl.indexOf("://");
            if (idx > 0) {
                connectUrl = connectUrl.substring(idx + 3);
            }

```
![image](https://user-images.githubusercontent.com/29959338/151911406-8c8940dd-902c-4a3d-a02a-b3a4c1fed43a.png)

![image](https://user-images.githubusercontent.com/29959338/151911486-8aa00604-397b-4adf-9528-c39aa5686870.png)

Reply:
@tadayosi see last commit. I created a new function `toJolokiaProxyUrl` to work around the glitch.
This now assumes that everything will go through the proxy, which I believe is fine. 
Let me know if this solution is acceptable.